packages/guides-restructured-text/src/RestructuredText/Parser/Productions/DirectiveContentRule.php

```diff
@@ -0,0 +1,37 @@
+<?php
+
+declare(strict_types=1);
+
+namespace phpDocumentor\Guides\RestructuredText\Parser\Productions;
+
+use phpDocumentor\Guides\Nodes\CompoundNode;
+use phpDocumentor\Guides\Nodes\Node;
+use phpDocumentor\Guides\RestructuredText\Nodes\DirectiveContentNode;
+use phpDocumentor\Guides\RestructuredText\Parser\BlockContext;
+
+/** @implements Rule<DirectiveContentNode> */
+class DirectiveContentRule implements Rule
+{
+    public function __construct(private readonly RuleContainer $bodyElements)
+    {
+    }
+
+    public function applies(BlockContext $blockContext): bool
+    {
+        return true;
+    }
+
+    public function apply(BlockContext $blockContext, CompoundNode|null $on = null): Node|null
+    {
+        $node = new DirectiveContentNode([]);
```

Comment:
We could just use the collectionnode here as the directivecontentnode is not an existing node. It's just for parsing collection. Most directives are directly copying the children from this class.

Reply:
Good idea!